python/xorbits/core/data.py

```diff
@@ -253,6 +253,33 @@ def __repr__(self):
             run(self)
             return self.data.__repr__()
 
+    def __int__(self):
+        from .execution import run
+
+        if (
+            self.data.data_type == DataType.tensor
+            and self.data._mars_entity.shape == ()
+            and self.data._mars_entity.dtype == "int"
+        ):
+            run(self)
+            return self.to_numpy()
+        elif self.data.data_type == DataType.object_ and self.to_object().isnumeric():
```

Comment:
`self.to_object()` will trigger execution, we don't need to check if it is numeric.